src/toxcore/dht.rs

```diff
@@ -319,10 +319,10 @@ impl ToBytes for IpAddr {
         debug!(target: "IpAddr", "Serializing IpAddr to bytes.");
         trace!(target: "IpAddr", "With IpAddr: {:?}", self);
         match *self {
-            IpAddr::V4(a) => a.octets().iter().map(|b| *b).collect(),
+            IpAddr::V4(a) => a.octets().iter().cloned().collect(),
```

Comment:
>`cloned()`
>Creates an iterator which clone()s all of its elements.

Doesn't `map(|b| *b)` simply de-reference values of an iter, while cloning unnecessarily allocates memory?